### PR TITLE
171 freedomline test coverage

### PR DESCRIPTION
**Pull request by manuelkoso**
Closing #171

### CODE REVIEW

**Comment by peiva-git** [comment]
Did some minor refactoring, mainly name changing.

What about making the `FreedomLine(Board<? extends Piece> board)` constructor `private`? Since it's never used outside the `FreedomLine` class: this way, we may also be able to declare the `color` and `direction` fields `@NotNull`, which would make more sense since we're not even testing the case where the `color` is `null` and the `color` can only be black or white anyway.

File: core/test/java/game/providers/FreedomLineProviders.java [hunk 1]
```diff
@@ -31,47 +58,109 @@ public class FreedomLineProviders {
     public static @NotNull Stream<Arguments> provideSetOfPositions() {
         return Stream.of(
                 Arguments.of(
-                        new TreeSet<>(Set.of(
+                        new LinkedHashSet<>(List.of(
```
Comment (peiva-git) [inline core/test/java/game/providers/FreedomLineProviders.java:61]: Why are you providing a set of lists? Wouldn't be `Set.of(...)` more than enough? Considering the `addPositions()` method only requires a `Set`

**Comment by manuelkoso** [comment]
You can give a try, but the direction must be used only for internal usage. The user must not set the direction. Also, I didn't get the colour thing.

**Review by peiva-git — CHANGES_REQUESTED** [review]
See the attached comments

**Comment by manuelkoso** [comment]
Ok, I got what you think. You can give a try

**Comment by manuelkoso** [comment]
Note that the direction is even set to null if there is only one stone

File: core/test/java/game/providers/FreedomLineProviders.java (hunk 1 above)
Comment (manuelkoso) [reply core/test/java/game/providers/FreedomLineProviders.java:61]: Nope, because I need that the input positions are insertion ordered.

**Comment by manuelkoso** [comment]
Have you already started to do that? Because otherwise, I will try to do it now.

File: core/test/java/game/providers/FreedomLineProviders.java (hunk 1 above)
Comment (peiva-git) [reply core/test/java/game/providers/FreedomLineProviders.java]: But then the `addPositions()` method signature is incorrect! The method needs a `SortedSet` in order to work properly

Comment (manuelkoso) [reply core/test/java/game/providers/FreedomLineProviders.java]: Note that with your last commit the test coverage is not more 100%

**Comment by peiva-git** [comment]
I just pushed the latest version. It should be clearer from the test methods what the desired behavior is for the `FreedomLine` class. 

> But then the addPositions() method signature is incorrect! The method needs a SortedSet in order to work properly

This is an example of what I mean. Another one is the `public` costructor accepting only the board parameter. I get the idea of just creating a new line instance and then filling it with `Position`s, but the class' correct usage should be clearer

**Comment by manuelkoso** [comment]
The `addPositions` signature is correct. I need to use an order set only for testing purposes.

File: core/test/java/game/providers/FreedomLineProviders.java (hunk 1 above)
Comment (peiva-git) [reply core/test/java/game/providers/FreedomLineProviders.java]: Then it should be clearer what the method does: when does it fail? When does it succeed? Why does the test pass with ordered positions, but fail with unordered ones?

**Comment by manuelkoso** [comment]
> I just pushed the latest version. It should be clearer from the test methods what the desired behavior is for the `FreedomLine` class.
> 
> > But then the addPositions() method signature is incorrect! The method needs a SortedSet in order to work properly
> 
> This is an example of what I mean. Another one is the `public` costructor accepting only the board parameter. I get the idea of just creating a new line instance and then filling it with `Position`s, but the class' correct usage should be clearer

As I told you with your last commit the test coverage is no longer 100%

File: core/test/java/game/providers/FreedomLineProviders.java (hunk 1 above)
Comment (manuelkoso) [reply core/test/java/game/providers/FreedomLineProviders.java]: I will create for each direction a different method provider in order to improve the readability.  Hope it helps

**Comment by manuelkoso** [comment]
I refactored the providers and added some docs. I hope that is clearer now. Let me know.
Now I will work on the constructor.

**Comment by manuelkoso** [comment]
I did it. Let me know what you think, but it seems pretty nice now :)

**Review by peiva-git — APPROVED** [review]
Now it's much clearer! I refactored the docs to better reflect how (I think) the method works. I'll approve the request anyway, but you can check them out first and see how you like them

**Comment by manuelkoso** [comment]
Well done, it seems ok to me. Whenever you want, you can merge the branch
